homeassistant/components/history/__init__.py

```diff
@@ -396,7 +396,13 @@ def entity_filter(self):
         if not excludes and includes:
             return not_(or_(*excludes))
 
-        return or_(*includes) & not_(or_(*excludes))
+        if includes and excludes:
+            return or_(*includes) & not_(or_(*excludes))
+
+        if includes:
+            return or_(*includes)
```

Comment:
Isn't this already handled above, in the non changed code?

Reply:
hmm yeah, I wonder why I still got the warning then 🤔